File: RNApp/ios/Bridge/Bridge.m
//
//  Bridge.m
//  RNApp
//
//  Created by Antoine Eddi on 11/4/20.
//

#import <React/RCTBridgeModule.h>
#import "React/RCTBridge.h"

@interface RCT_EXTERN_MODULE(Bridge, NSObject)
   RCT_EXTERN_METHOD(test:(RCTPromiseResolveBlock)resolve
                     reject:(RCTPromiseRejectBlock)reject)
@end

File: RNApp/ios/Bridge/Bridge.swift
//
//  Bridge.swift
//  RNApp
//
//  Created by Antoine Eddi on 11/4/20.
//

import Foundation
import Cgo

@objc(Bridge)
class Bridge: NSObject {
  @objc func test(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
    let result = CgoTest("All your base are belong to us")
    resolve(result)
  }
}
